src/main/java/io/ebeaninternal/server/query/CQuery.java

```diff
@@ -317,6 +317,9 @@ private boolean prepareBindExecuteQueryWithOption(boolean forwardOnlyHint) throw
         return false;
       }
 
+      // Call to native method nanoTime() while locked causes an expensive context switch.
+      // startNano is used only in updateExectuionStatistics, which itself is called in various places.
+      // Is this a problem?
       startNano = System.nanoTime();
```

Comment:
The usage of `System.currentTimeMillis()` is it possible ? Yes you always have the context switching, but `currentTimeMillis` is faster than `nanoTime`.

Basically, `System.currentTimeMillis()` just reads a global variable maintained by Windows (which is what it has low granularity), whereas `System.nanoTime()` actually has to do IO operations.